src/web/middleware/jira/jira-iframe-symmetric-jwt-authentication-middleware.ts

```diff
@@ -0,0 +1,44 @@
+import { fromExpressRequest } from 'atlassian-jwt';
+import type { NextFunction, Request, RequestHandler, Response } from 'express';
+
+import { isString } from '../../../common/string-utils';
+import { jiraIframeOrServerToServerSymmetricJwtTokenVerifier } from '../../../infrastructure/jira/inbound-auth';
+import { UnauthorizedResponseStatusError } from '../../errors';
+
+/**
+ * Authenticates requests using an iframe symmetric JWT token received in a query param.
+ *
+ * In case of successful authentication, `connectInstallation` and `atlassianUserId` are set in locals.
+ *
+ * @remarks
+ * Context JWT tokens are sent in a query param when loading an iframe
+ *
+ * @see https://developer.atlassian.com/cloud/jira/platform/understanding-jwt-for-connect-apps/#types-of-jwt-token
+ * @see https://community.developer.atlassian.com/t/action-required-atlassian-connect-vulnerability-allows-bypass-of-app-qsh-verification-via-context-jwts/47072
+ */
+export const jiraIframeSymmetricJwtAuthenticationMiddleware: RequestHandler = (
```

Comment:
**note:** Represents a replacement for the deleted `jiraContextSymmetricJwtFromQueryAuthenticationMiddleware` (introduced within https://github.com/atlassian-labs/figma-for-jira/pull/208) as the correct way to authenticate requests from iframes.

The new middleware uses a different verifier (`jiraIframeOrServerToServerSymmetricJwtTokenVerifier` instead of `jiraContextSymmetricJwtFromQueryAuthenticationMiddleware`) and apples minor fixes to how a request is handled.

However, the middleware is **not** currently used and can be safely removed if the team agrees on serving static content without authentication. Please, see other notes for more detail.